core/new-gui/src/app/workspace/component/visualization-panel-content/visualization-panel-content.component.ts

```diff
@@ -0,0 +1,87 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import { MAT_DIALOG_DATA} from '@angular/material/dialog';
+import * as c3 from 'c3';
+import {PrimitiveArray} from 'c3';
+
+@Component({
+  selector: 'texera-visualization-panel-content',
+  templateUrl: './visualization-panel-content.component.html',
+  styleUrls: ['./visualization-panel-content.component.scss']
+})
+export class VisualizationPanelContentComponent implements OnInit {
+  table: object[];
+  columns: string[] = [];
+  map: Map<string, string[]>;
+ 
+
+  constructor(@Inject(MAT_DIALOG_DATA) public data: any) {
+    this.table = data.table;
+    this.map = new Map<string, string[]>();  
+  }
+
+  getKeys(map: Map<string, string[]>): string[] {
+    return Array.from(map.keys());
+}
+  ngOnInit() {
+ 
+    this.columns = Object.keys(this.table[0]).filter(x => x !== '_id');
+   
+    for (let column of this.columns) {
+      
+      let rows: string[] = [];
+     
+      for (let row of this.table) {
+        rows.push(String((row as any)[column]));
+      }
+     
+      this.map.set(column, rows);
+      
+    }
+
+   
+  }
+
+  ngAfterViewInit() {
+    this.onClickGenerateChart();
+  }
+
+  async onClickGenerateChart() {
```

Comment:
any reason this function needs to be async?

Reply:
I will remove it.